src/shogun/lib/DynamicArray.h

```diff
@@ -316,17 +330,35 @@ template <class T> class CDynamicArray :public CSGObject
 		 */
 		inline bool insert_element(T e, int32_t index)
 		{
-			return m_array.insert_element(e, index);
+			try
+			{
+				m_array.insert(m_array.begin() + index, e);
+
+				num_elements++;
+				return true;
+			}
+			catch (const std::bad_alloc&)
+			{
+				return false;
+			}
 		}
 
 		/** append array element to the end of array
 		 *
 		 * @param e element to append
 		 * @return if setting was successful
 		 */
-		inline bool append_element(T e)
+		inline void append_element(T e)
```

Comment:
i wonder why do you change the return type of the function?
and you kept the doxygen comment which is then now confusing.... please just keep the return value `bool`